org.eclipse.m2e.pde.connector/src/org/eclipse/m2e/pde/connector/TychoPackagingsConfigurator.java

```diff
@@ -0,0 +1,131 @@
+import org.eclipse.m2e.core.project.configurator.ProjectConfigurationRequest;
+import org.eclipse.pde.ds.internal.annotations.DSAnnotationVersion;
+import org.eclipse.pde.internal.core.natures.PDE;
+import org.eclipse.pde.internal.core.util.CoreUtility;
+import org.osgi.framework.Version;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("restriction")
+public class TychoPackagingsConfigurator extends AbstractProjectConfigurator {
+
+    private static final String TYCHO_GROUP_ID = "org.eclipse.tycho";
+    private static final String TYCHO_DS_PLUGIN_ARTIFACT_ID = "tycho-ds-plugin";
+    private static final String GOAL_DECLARATIVE_SERVICES = "declarative-services";
+
+    private static final Logger log = LoggerFactory.getLogger(TychoPackagingsConfigurator.class);
+   
+    @Override
+    public void configure(ProjectConfigurationRequest request, IProgressMonitor monitor) throws CoreException {
+        MavenProject mavenProject = request.mavenProject();
+        IProject project = request.mavenProjectFacade().getProject();
+
+        String packaging = mavenProject.getPackaging();
+        if ("eclipse-plugin".equals(packaging) || "eclipse-test-plugin".equals(packaging)) {
+            PDEProjectHelper.configurePDEBundleProject(project, mavenProject, monitor);
+            applyDsConfiguration(request, monitor);
+        } else if ("eclipse-feature".equals(packaging)) {
+            // see org.eclipse.pde.internal.ui.wizards.feature.AbstractCreateFeatureOperation
+            if (!project.hasNature(PDE.FEATURE_NATURE)) {
+                CoreUtility.addNatureToProject(project, PDE.FEATURE_NATURE, monitor);
+            }
+        }
+    }
+
+    void applyDsConfiguration(ProjectConfigurationRequest request, IProgressMonitor monitor) throws CoreException {
+        List<MojoExecution> mojoExecutions = getTychoDsPluginMojoExecutions(request, monitor);
+        if (mojoExecutions.isEmpty()) {
+            return;
+        }
+        if (mojoExecutions.size() > 1) {
+            log.error("Found more than one execution for plugin {}:{} and goal {}, only consider configuration of first one", TYCHO_GROUP_ID, TYCHO_DS_PLUGIN_ARTIFACT_ID, GOAL_DECLARATIVE_SERVICES);
+        }
+        // first mojo execution is relevant
+        Xpp3Dom dom = mojoExecutions.get(0).getConfiguration();
+        // apply PDE configuration for DS
+        IEclipsePreferences prefs = new ProjectScope(request.mavenProjectFacade().getProject())
+                .getNode(org.eclipse.pde.ds.internal.annotations.Activator.PLUGIN_ID);
+        Xpp3Dom dsEnabled = dom.getChild("enabled");
+        final boolean isDsEnabled;
+        if (dsEnabled != null && !dsEnabled.getValue().isEmpty()) {
+            isDsEnabled = Boolean.valueOf(dsEnabled.getValue());
+            prefs.putBoolean(org.eclipse.pde.ds.internal.annotations.Activator.PREF_ENABLED, isDsEnabled);
+        } else {
+            isDsEnabled = false;
+        }
```

Comment:
```suggestion
		boolean isDsEnabled = false;
		if (dsEnabled != null && !dsEnabled.getValue().isEmpty()) {
			isDsEnabled = Boolean.valueOf(dsEnabled.getValue());
			prefs.putBoolean(org.eclipse.pde.ds.internal.annotations.Activator.PREF_ENABLED, isDsEnabled);
		}
```
No need for an extra else-block.

Reply:
Done in https://github.com/eclipse-m2e/m2e-core/pull/967/commits/b30f5013acbf8e7e38c8e6166de8e9ac876f278a.